src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py

```diff
@@ -303,6 +304,20 @@ def flexible_server_version_upgrade(cmd, client, resource_group_name, server_nam
                        "the option to select version 12 will be removed in the near future. "
                        "We recommend selecting PostgreSQL 13 or a later version for "
                        "all future operations.")
+    
+    list_server_capability_info = get_postgres_server_capability_info(cmd, resource_group_name, server_name)
+    eligible_versions = list_server_capability_info['supported_server_versions'][str(current_version)]
+    if version not in eligible_versions:
+        error_message = ""
+        if len(eligible_versions) > 0:
+            error_message = "The version to upgrade to must be one of the supported versions for the current server version from this list: {}. ".format(eligible_versions)
+        else:
+            error_message = "Your server is up-to-date with the latest version. "
```

Comment:
"Your server is up-to-date with the latest version. " => "Server is running version {0}. It cannot be upgraded to any higher version."